Cannot find module by Relative Path on Windows

Hello,

I'm trying to use a node library that uses threads.js to load a Worker with a relative path (`new Worker('./file.js')`). It works fine on Linux, but fail on Windows with `Error: Cannot find module '\\H:\correct\absolute\path\to\file.js'`
It seems to me the culprits are the `\\` remaining from the file protocol, Windows likes to add what seems to be an arbitrary number of them :D
https://github.com/andywer/threads.js/blob/8d741aa51e04ac27ae2e3cd7070d558b6babfa1e/src/master/implementation.node.ts#L72
In my case the raw path begins with `file:///H:/`, patching the regex to consume additional slashes allow the worker to be loaded correctly, but this obviously won't work in other cases.
I also tried to workaround the problem by using an absolute path, but it looks like it's still treated as if it was an relative one (trying to load `H:\correct\absolute\path\to\H:\correct\absolute\path\to\file.js`)

As a side note, as I couldn't make sense of `__non_webpack_require__` by looking at the source, this is a node application written in typescript, no webpack involved.

--- a/src/master/implementation.node.ts
+++ b/src/master/implementation.node.ts
@@ -5,6 +5,7 @@ import getCallsites, { CallSite } from "callsites"
 import { EventEmitter } from "events"
 import { cpus } from 'os'
 import * as path from "path"
+import { fileURLToPath } from "url";
 import {
   ImplementationExport,
   ThreadsWorkerOptions,
@@ -69,7 +70,10 @@ function rebaseScriptPath(scriptPath: string, ignoreRegex: RegExp) {
   })
 
   const rawCallerPath = parentCallSite ? parentCallSite.getFileName() : null
-  const callerPath = rawCallerPath ? rawCallerPath.replace(/^file:\//, "") : null
+  let callerPath = rawCallerPath ? rawCallerPath : null;
+  if (callerPath && callerPath.startsWith('file:')) {
+    callerPath = fileURLToPath(callerPath);
+  }
   const rebasedScriptPath = callerPath ? path.join(path.dirname(callerPath), scriptPath) : scriptPath
 
   return rebasedScriptPath

